lib/ansible/modules/cloud/amazon/efs_facts.py

```diff
@@ -348,10 +361,14 @@ def main():
     if tags:
         file_systems_info = [item for item in file_systems_info if has_tags(item['tags'], tags)]
 
+    file_systems_info = connection.get_mount_targets_data(file_systems_info)
+
     if targets:
         targets = [(item, prefix_to_attr(item)) for item in targets]
         file_systems_info = [item for item in file_systems_info if has_targets(item['mount_targets'], targets)]
 
+    file_systems_info = connection.get_security_groups_data(file_systems_info)
```

Comment:
Right now a task like:
```
    - name: get efs facts using targets
      efs_facts:
        profile: shertel
        region: us-east-1
        targets:
          - 'sg-5e27db2f'
```
doesn't return anything because file_systems_info is being overwritten by the list comprehension above. This line should go above `if targets:`. Looks good to merge besides that.

Reply:
Good catch